backend/lib/endpoints/posts.js

```diff
@@ -162,8 +165,10 @@ async function routes(app) {
       );
 
       if (postsErr) {
-        req.log.error("Failed requesting posts", { postsErr });
+        req.log.error(postsErr, "Failed requesting posts");
         throw app.httpErrors.internalServerError();
+      } else if (posts === null) {
+        throw app.httpErrors.notFound();
```

Comment:
I do have a question @robinv85  In what scenario is `posts === null`? I've been using postman to trigger this case, and even an empty db returns an empty array, never null

Reply:
ah yeah well I don't know if this can happen, I thought you considered that
I wasn't sure how the Mongoose API works in all cases, I think there are 2 cases that should be covered
- error occurs
- collection doesn't exist


So if `posts === null` only occurs in case of error, that was already covered
maybe could put them together like `if ( postErr || !posts ) { throw 500 } `

Reply:
I think we only have null for Mongoose `findById`, `findOne` , not `find` similar to most ORM/ODM libraries I think. So 

```
      } else if (posts === null) {
        return [];
      }
```

Is unnecessary but at the same time not harmful (might just cause confusion later) we can revisit with refactoring but I'm fine to merge.

Reply:
I'll create a ticket to follow up on these changes in the future. Thanks all!